src/net/sourceforge/kolmafia/session/RufusManager.java

```diff
@@ -535,14 +534,12 @@ public static void handleShadowRiftFight(MonsterData monster) {
           "shadow matrix" -> {
         // Bosses are NCs
         Preferences.setInteger("shadowRiftLastNC", currentTurns);
-        lastNC = currentTurns;
+        Preferences.resetToDefault("encountersUntilSRChoice");
       }
     }
 
     // All fights - including bosses - advance turns in zone
     currentTurns = Preferences.increment("shadowRiftTotalTurns", 1);
```

Comment:
Since you threw away the calculation that uses currentTurns, you do not need to set currentTurns.